rinex-cli/src/positioning/mod.rs

```diff
@@ -206,52 +211,21 @@ pub fn precise_positioning(ctx: &Context, matches: &ArgMatches) -> Result<(), Er
         panic!("High precision orbits (SP3) are unfortunately mandatory at the moment..");
     }
 
+    let mut orbit = RefCell::new(OrbitInterpolator::from_ctx(
```

Comment:
Is the `RefCell` here necessary?

Reply:
It is how I solved the following problem (haven't gone through your suggestions yet).
The RTK core uses function pointers (`Fn` closures) at instant `t` when it need some information to proceed.
It is a simple yet efficient method to make RTK agnostic of the end user (RINEX in our case).
One example of that is `Fn (t, sv)` to request results of position interpolation.
The "problem" is that Rust is not smart enough to understand this application is not multithreaded at the moment, and we're not allowed to capture the RinexContext (the data source, data provider, in our case) without troubles..

Reply:
> The RTK core uses function pointers (Fn closures) at instant t when it needs some information.

I see what you mean. I didn't test, but `gnss-rtk` might be able to take a `FnMut` instead of `Fn`. `resolve` takes `&mut self`, so that might work.

Reply:
Interesting, I did not know of FnMut